docs/src/pages/theming.mdx

```diff
@@ -0,0 +1,977 @@
+// @ts-check
+const { withUt } = require("@uploadthing/react/tw");
+
+/** @type {import('tailwindcss').Config} */
+module.exports = withUt({
+  // your config goes here
+});
+```
+
+<Callout type="info">
+  If you're not wrapping your config as shown above, you have to import our
+  stylesheet into your app. Otherwise, components will not look right.
+
+```ts filename="app/layout.tsx"
+import "@uploadthing/react/styles.css";
+
+// ...
+```
+
+</Callout>
+
+#### Style using the `className` prop
+
+`className` accepts any classes and will merge them using `tailwind-merge`,
+meaning you can pass any class you like and it will be applied correctly,
+overriding the default styles if necessary.
+
+The `withUt` wrapper adds custom variants that you can leverage to easily target
+different elements of the component and its state:
+
+<div className="mx-auto w-max mt-4">
+
+| Variant               | Description                                                       |
+| --------------------- | ----------------------------------------------------------------- |
+| `ut-button:`          | Applied to the button element.                                    |
+| `ut-allowed-content:` | Applied to the allowed content element.                           |
+| `ut-label:`           | Applied to the label element.                                     |
+| `ut-upload-icon:`     | Applied to the upload icon element.                               |
+| ---                   | ---                                                               |
+| `ut-readying:`        | Applied to the container element when the component is readying.  |
+| `ut-ready:`           | Applied to the container element when the component is ready.     |
+| `ut-uploading:`       | Applied to the container element when the component is uploading. |
+
+</div>
+
+<Callout>
+  If you're not using the `withUt` wrapper, the state variants can be applied
+  using `data-[state="..."]:`
+</Callout>
+
+These variants and classes can be used in conjunction with each other to make
+component match your design in the exact way you want.
+
+```tsx
+<UploadButton
+    className="mt-6 ut-button:bg-red-500 ut-button:ut-readying:bg-red-500/50"
+               |    └─ applied to `button`  └─ applied to `button` when `readying`
+               └─ applied to `container`
+/>
+```
+
+<div
+  className={twMerge(
+    "flex flex-col items-center justify-center gap-1",
+    "ut-button:bg-red-500 ut-button:ut-readying:bg-red-500/50 mt-6",
+  )}
+>
+  <label
+    className="relative flex h-10 w-36 cursor-pointer items-center justify-center overflow-hidden rounded-md bg-blue-600 text-white after:transition-[width] after:duration-500"
+    data-ut-element="button"
+  >
+    <input className="sr-only" disabled />
+    <p>Ready</p>
+  </label>
+  <div
+    className="h-[1.25rem] text-xs leading-5 text-white"
+    data-ut-element="allowed-content"
+  >
+    <p>Allowed content</p>
+  </div>
+</div>
```

Comment:
Can we not setup a mock endpoint & router and use the components from the react lib? Will make it more future proof

Reply:
How to handle demonstration of state-specific styles in this case? I mean `readying` state is available only during quite short amount of time after component mount. State `uploading` requires actually uploading files and also visible for short amount of time

Reply:
hmm - we could add a `__internal-state` prop or something to control the state (doesn't even have to be exposed as a type) and use that?

```mdx
<UploadButton __internal_state="ready" endpoint="foobar" />
<UploadButton __internal_state="readying" endpoint="foobar" />
<UploadButton __internal_state="uploading" endpoint="foobar" />
```

Reply:
I was thinking about something like that as well. I also think that if that state is used we actually should make components non interactive. I mean that we don't actually want users be able to upload anything. Also we don't need to actually fetch config and therefore we don't need exposing an API route. What do you think?